tests/test-sessions/test-sessions-common/src/main/java/org/eclipse/jetty/server/session/AbstractSessionDataStoreTest.java

```diff
@@ -272,6 +279,15 @@ public void testStoreObjectAttributes() throws Exception
 
             //Make an attribute that uses the proxy only known to the webapp classloader
             data.setAttribute("a", proxy);
+            
+            //Now make an attribute that uses a system class to store a webapp classes
```

Comment:
What about a the opposite direction?
A webapp class that uses system classes inside of it?
Eg: a webapp with google-collections with a Bag of DataSources?

Reply:
What about ...

```
Map<WebAppToken, GoogleBag<java.nio.FileSystem>> ?
```

Reply:
If the context class loader can see all the classes then it is not a problem. If the context class loader cannot see the contained classes then it is a problem.... but applications cannot create such an attribute... Only the container can.... So we just don't do that... If we need something like that then we have to create a container visible only container (eg extend at list).